src/types/products.ts

```diff
@@ -1,10 +1,21 @@
 import { ProductMeta, Products } from '@hashicorp/platform-product-meta'
+import { NavigationHeaderItem } from 'components/navigation-header/types'
+import { MenuItem } from 'components/sidebar'
 
 type ProductSlug = Exclude<Products, 'hashicorp'> | 'hcp' | 'sentinel'
 
+/**
+ * TODO: put basePaths in a separate interface that extends the Product one?
+ */
 interface Product extends ProductMeta {
+  basePaths?: string[]
+  navigationHeaderItems?: NavigationHeaderItem[]
+  sidebar?: {
+    landingPageNavData: MenuItem[]
+    resourcesNavData: MenuItem[]
+  }
   slug: ProductSlug
-  url: string
+  url?: string
```

Comment:
I think we also need a `name` property here - used in the waypoint product landing `server.ts`, and also in the docs layout `server.ts`. Alternately I guess we could always derive the product name from the slug? Though I worry about edge cases with eg. HCP or sub-products, where we might want to pass a specific product slug but have more flexibility in the displayed name (eg `HCP` / `HashiCorp Cloud Platform`).

Reply:
Ah, yes! `name` is very necessary just slipped the mind to add it so far. 😊

Names are actually currently derived here: https://github.com/hashicorp/dev-portal/blob/assembly-ui-v1/config/products.ts#L40. I'm pretty sure I referenced some existing code for that but I don't remember where it is.